its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/BaseDirTest.java

```diff
@@ -98,37 +97,51 @@ void testAzureFunctions_WithWrongBaseDirectory_AnalysisSucceeds() throws IOExcep
   }
 
   @Test
-  void testCSharpSharedFileWithOneProjectUsingProjectBaseDirAbsolute() throws IOException {
-    runCSharpSharedFileWithOneProjectUsingProjectBaseDir(
-      projectDir -> {
-        try {
-          return projectDir.toRealPath(LinkOption.NOFOLLOW_LINKS).toString();
-        } catch (IOException e) {
-          e.printStackTrace();
-        }
-        return null;
-      });
+  void projectBaseDir_Absolute() {
+    var context = AnalysisContext.forServer("CSharpSharedFileWithOneProject");
+    assertThat(context.projectDir.toString()).doesNotContain("~"); // Scaffolding does .toRealPath to convert it from DOS 8.3 format
```

Comment:
This assertion is testing `AnalysisContext` behavior. The place where this is done is fairly random. Do we have a better place? Maybe in the "EmpytProject" that get's analyzed on "Init"?